### PR TITLE
Fix create project from db not loading dbs when disconnected

### LINKED ISSUE #19063: "Create Project from Database" fails to load DBs from disconnected node
1. Right click on a disconnected server node
2. Select "Create Project from Database"

The dialog opens up but the Database dropdown is empty and this error is listed in the console 

```
console.ts:137 [Extension Host] TypeError: Cannot read property 'databaseNames' of undefined    at p.$listDatabases (vscode-file://vscode-app/c:/Users/chgagnon/AppData/Local/Programs/Azure%20Data%20Studio%20-%20RC1/resources/app/out/vs/workbench/workbench.desktop.main.js:2311:18212)    at processTicksAndRejections (internal/process/task_queues.js:93:5) 
```

### PR DESCRIPTION
**Pull request by kisantia**
This PR fixes #19063.

Automatically connect if possible:
autoConnect2

Open connection dialog if can't automatically connect (updated from first iteration to open connection dialog first, before opening create project from db dialog):
openConnDialogFirst

### CODE REVIEW

File: extensions/sql-database-projects/src/dialogs/createProjectFromDatabaseDialog.ts [hunk 1]
```diff
@@ -54,6 +54,19 @@ export class CreateProjectFromDatabaseDialog {
...
+			const connections = await azdataType.connection.getConnections(true);
+			let connection = connections.find(c => c.connectionId === this.profile!.id);
+
+			if (!connection) {
+				// if the connection clicked on isn't currently connected, try to connect
+				const result = await azdataType.connection.connect(this.profile, true, false);
+
+				if (!result.connected) {
+					// if can't connect automatically, open connection dialog with the info from the profile
+					await azdataType.connection.openConnectionDialog(undefined, this.profile);
```
Comment (Benjin) [inline extensions/sql-database-projects/src/dialogs/createProjectFromDatabaseDialog.ts:66]: Not sure how I feel about this; gif looks a little jarring to open one dialog, pause for a moment, then open another one, losing some context along the way.

I think it'd feel smoother if did nothing if it couldn't connect, or maybe gave it an error box around the field in the dialog saying it couldn't connect.

Comment (kisantia) [reply extensions/sql-database-projects/src/dialogs/createProjectFromDatabaseDialog.ts]: Yeah makes sense that it was a little jarring opening the connection dialog after opening the create project dialog. Does opening the connection dialog (if auto connecting didn't work) first look better?
openConnDialogFirst

If not, I can remove opening the connection dialog at all if the connection couldn't automatically be made, but then the user would have to open the connection dialog themselves anyway if they wanted to create a project from a database on this connection.

File: extensions/sql-database-projects/src/dialogs/createProjectFromDatabaseDialog.ts [hunk 2]
```diff
@@ -50,11 +50,37 @@ export class CreateProjectFromDatabaseDialog {
...
+					// if can't connect automatically, open connection dialog with the info from the profile
+					const connection = await azdataType.connection.openConnectionDialog(undefined, this.profile);
+					connected = !!connection;
+
+					// update these fields if connection was successful, to ensure they match the connection made
+					if (connected) {
+						this.profile.id = connection.connectionId;
+						this.profile.databaseName = connection.options['databaseName'];
+						this.profile.serverName = connection.options['server'];
+						this.profile.userName = connection.options['user'];
```
Comment (udeeshagautam) [inline extensions/sql-database-projects/src/dialogs/createProjectFromDatabaseDialog.ts:73]: this is the exhaustive list - any chance we could miss copying something from connection object?

**Review by udeeshagautam — APPROVED** [review]
Looks good to me with new gifs.

File: extensions/sql-database-projects/src/dialogs/createProjectFromDatabaseDialog.ts (hunk 2 above)
Comment (kisantia) [reply extensions/sql-database-projects/src/dialogs/createProjectFromDatabaseDialog.ts:73]: these are only used for displaying the connection in the dialog. The connection id is what actually gets used for creating the project

**Pull request merged**